File: backend/micro_epoweredmove_poi_chargingstation/src/main/java/com/kosdiam/epoweredmove/models/ChargingStation.java
package com.kosdiam.epoweredmove.models;

import java.io.Serializable;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class ChargingStation implements Serializable {

    private String id;
    private Float pricePerKwh;
    private Boolean autoAccept; //reserve without owner permission
    private Boolean barcodeEnabled;
    private Boolean apiEnabled;
    private String imageId;
}

File: frontend/E-PoweredMoveAndroidApp/app/src/main/java/com/kosdiam/epoweredmove/models/ChargingStation.kt
package com.kosdiam.epoweredmove.models

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class ChargingStation(
    val id: String,
    val pricePerKwh: Float,
    val autoAccept: Boolean,
    val barcodeEnabled: Boolean,
    val apiEnabled: Boolean,
    val imageId: String?,
) : Parcelable{
    constructor(): this("", 0f, true, true, true, "")
}
